src/net.cpp

```diff
@@ -2836,22 +2840,22 @@ std::vector<CAddress> CConnman::GetAddresses(CNode& requestor, size_t max_addres
     return cache_entry.m_addrs_response_cache;
 }
 
-bool CConnman::AddNode(const std::string& strNode)
+bool CConnman::AddNode(const std::string& strNode, ConnectionType conn_type)
```

Comment:
Should this assert or return false if `conn_type != ConnectionType::MANUAL && conn_type != ConnectionType::MANUAL_BLOCK_RELAY`? What would it mean to have a non-manual connection type in the `m_added_nodes` container?

If this is changed to an assert you'd need to also update the test in fuzz/connman.

Reply:
I added an assert and updated the fuzz test to comply with that.